### DISCUSSION TITLE
Async templates?

### DISCUSSION THREAD

**Discussion by heggemsnes** [sanity-io/sanity#4191] [Sanity Studio Support]
Hi!

I'm looking into the new initial value templates api and wondering if it's possible or planned to be possible to make the templates callback able to be async?

My use case is a "quiz builder" and would be nice to have the option to create a new question based on some defined templates.

**Comment by jackieang-braze** [comment]
Would also like to know about this!

**Comment by eivindml** [comment]
This would be very handy. I want to be able to set a field values based on a global configuration doc.
